Streetcode/Streetcode.WebApi/Utils/WebParsingUtils.cs

```diff
@@ -1,73 +1,341 @@
 ﻿using System.IO.Compression;
+using System.Linq;
+using System.Net;
+using System.Runtime.Serialization.Json;
 using System.Text;
-using OfficeOpenXml;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Microsoft.Net.Http.Headers;
+using Newtonsoft.Json;
 
-namespace Streetcode.WebApi.Utils;
-
-public class WebParsingUtils
+namespace Streetcode.WebApi.Utils
 {
-    public static async void ParseZipFileFromWeb(object? state)
+    public class WebParsingUtils
     {
-        var projRootDirectory = Directory.GetParent(Environment.CurrentDirectory)?.FullName!;
+        public static async Task DownloadAndExtractAsync(string fileUrl, string zipPath, string extractTo, CancellationToken cancellationToken, ILogger logger = null)
+        {
+            if (string.IsNullOrEmpty(fileUrl))
+            {
+                throw new ArgumentException("FileUrl cannot be null or empty", nameof(fileUrl));
+            }
+
+            if (!Uri.IsWellFormedUriString(fileUrl, UriKind.Absolute))
+            {
+                throw new ArgumentException("Invalid FileUrl", nameof(fileUrl));
+            }
```

Comment:
These two checks may be merged into one.

Reply:
done